parquet-mr-tests/src/test/java/parquet/compat/test/TestBackwardsCompatibility.java

```
@@ -0,0 +1,108 @@
+        } catch (IOException ioe) {
+          LOG.info(ioe.getMessage());
+          LOG.info("Please run the main method of this Test class to " +
+          		"generate the required files");
+          Assert.fail();
+        }
+        File csvVersionedTestFile = Utils.getCsvTestFile(
+            versionParquetFile.getName()+"."+version, true);
+        
+        Utils.convertParquetToCSV(versionParquetFile, csvVersionedTestFile);
+        
+        Utils.verify(originalCsvFile, csvVersionedTestFile);
+      }
+    }
+  }
+  
+  @Test
+  public void testReadWriteCompatibility() throws IOException {
+    // convert original csv into parquet
+    // convert parquet back into csv
+    // diff converted csv with original csv
+    
+    File[] csvFiles = Utils.getAllOriginalCSVFiles();
+    for (File csvFile : csvFiles) {
+      File parquetTestFile = Utils.getParquetTestFile(
+          Utils.getFileNamePrefix(csvFile)+".java.readwrite.", true);
+      Utils.convertCsvToParquet(csvFile, parquetTestFile);
+      
+      File csvTestFile = Utils.getCsvTestFile(
+          Utils.getFileNamePrefix(csvFile)+".java.readwrite.", true);
+      Utils.convertParquetToCSV(parquetTestFile, csvTestFile);
+      
+      Utils.verify(csvTestFile, csvFile);
+    }
+  }
+  
+  /**
+   * This is a utility method to generate parquet files
+   * for a given release version. Files generated by this utility
+   * should be committed to git.
+   * 
+   * @param args
+   */
+  public static void main(String[] args) {
+    final File[] csvFiles = Utils.getAllOriginalCSVFiles();
+    
+    try {
+      for (File csvFile : csvFiles) {
+        File parquetVersionedFile = Utils.getParquetVersionedFile(Utils.getFileNamePrefix(csvFile), 
+            CURRENT_VERSION, false);
+        Utils.convertCsvToParquet(csvFile, parquetVersionedFile);
+        System.out.println(parquetVersionedFile.getName());
+      }
+    } catch (IOException ioe) {
+      System.err.println(ioe.getMessage());
```

just let the exception propagate up.
public static void main(String[] args) throws IOException